packages/ra-core/src/auth/usePermissions.ts

```diff
@@ -41,13 +42,19 @@ const usePermissions = <Permissions = any, Error = any>(
     }
 ) => {
     const authProvider = useAuthProvider();
+    const logoutIfAccessDenied = useLogoutIfAccessDenied();
 
     const result = useQuery(
         ['auth', 'getPermissions', params],
         authProvider
             ? () => authProvider.getPermissions(params)
             : async () => [],
-        queryParams
+        {
+            ...queryParams,
+            onError: error => {
+                logoutIfAccessDenied(error);
+            },
```

Comment:
You should call original `onError` in this provided `onError`.

Besides, I recommend logging the error in dev mode, to help the developer understand what is happening, just like we do here: https://github.com/marmelab/react-admin/blob/next/packages/ra-core/src/dataProvider/useDataProvider.ts#L114

Reply:
Actually it would be a breaking change to not allow overriding the `onError`